tests/antares/services/api_services/test_area_api.py

```diff
@@ -327,3 +329,97 @@ def test_create_misc_gen_success(self):
+            )
+        ]
+        json_st_storage = [
+            json.loads(
+                """{"id":"test_storage","group":"Pondage","name":"test_storage","injectionNominalCapacity":0,"withdrawalNominalCapacity":0,"reservoirCapacity":0,"efficiency":1,"initialLevel":0.5,"initialLevelOptim":false,"enabled":"true"}"""
+            )
+        ]
+        json_properties = json.loads(
+            """{"energyCostUnsupplied":0,"energyCostSpilled":0,"nonDispatchPower":true,"dispatchHydroPower":true,"otherDispatchPower":true,"filterSynthesis":["daily","monthly","weekly","hourly","annual"],"filterByYear":["daily","monthly","weekly","hourly","annual"],"adequacyPatchMode":"outside"}"""
+        )
+
+        with requests_mock.Mocker() as mocker:
+            mocker.get(ui_url, json=json_ui)
+            mocker.get(url_thermal, json=json_thermal)
+            mocker.get(url_renewable, json=json_renewable)
+            mocker.get(url_st_storage, json=json_st_storage)
+            mocker.get(url_properties_form, json=json_properties)
+
+            area_api = AreaApiService(self.api, study_id_test)
+            area_list = area_api.read_areas()
+            area_ui = area_api.craft_ui(url + "?type=AREA&ui=true", "zone")
+
+            thermal_id = json_thermal[0].pop("id")
+            thermal_name = json_thermal[0].pop("name")
+            renewable_id = json_renewable[0].pop("id")
+            renewable_name = json_renewable[0].pop("name")
+            storage_id = json_st_storage[0].pop("id")
+            storage_name = json_st_storage[0].pop("name")
+
+            thermal_props = ThermalClusterProperties(**json_thermal[0])
+            thermal_cluster = ThermalCluster(self.area_api.thermal_service, thermal_id, thermal_name, thermal_props)
+            renewable_props = RenewableClusterProperties(**json_renewable[0])
+            renewable_cluster = RenewableCluster(
+                self.area_api.renewable_service, renewable_id, renewable_name, renewable_props
+            )
+            storage_props = STStorageProperties(**json_st_storage[0])
+            st_storage = STStorage(self.area_api.storage_service, storage_id, storage_name, storage_props)
+
+            area_test = Area(
+                "zone",
+                self.area_api,
+                self.area_api.storage_service,
+                self.area_api.thermal_service,
+                self.area_api.renewable_service,
+                thermals={thermal_id: thermal_cluster},
+                renewables={renewable_id: renewable_cluster},
+                st_storages={storage_id: st_storage},
+                properties=json_properties,
+                ui=area_ui,
+            )
+
+            area_test_list = []
+            area_test_list.append(area_test)
+
+            assert len(area_list) == len(area_test_list)
```

Comment:
It's nice to test this but couldn't you simply test that the 2 list are the same ?
Also we often name `actual` what we receive and `expected` what's expected to help readability. You could do it here